core/src/json_schema.rs

```diff
@@ -282,6 +363,297 @@ pub mod simplified {
+            #[test]
+            fn $func_name() {
+                let res = $input.remove_multiple().unwrap();
+
+                assert_eq!(res, $expected);
+            }
+        };
+    }
+
+    // Testing TryFrom<json_schema::Schema> for json_schema::simplified::Atomic
+    mod remove_multiple {
+        use crate::{
+            json_schema::{BsonType, BsonTypeName, Items, Schema},
+            map,
+        };
+
+        remove_multiple_test!(
+            multiple_atomic,
+            expected = Schema {
+                bson_type: None,
+                properties: None,
+                required: None,
+                additional_properties: None,
+                items: None,
+                any_of: Some(vec![
+                    Schema {
+                        bson_type: Some(BsonType::Single(BsonTypeName::Int)),
+                        properties: None,
+                        required: None,
+                        additional_properties: None,
+                        items: None,
+                        any_of: None,
+                    },
+                    Schema {
+                        bson_type: Some(BsonType::Single(BsonTypeName::Null)),
+                        properties: None,
+                        required: None,
+                        additional_properties: None,
+                        items: None,
+                        any_of: None,
+                    }
+                ]),
+            },
+            input = Schema {
+                bson_type: Some(BsonType::Multiple(vec![
+                    BsonTypeName::Int,
+                    BsonTypeName::Null
+                ])),
+                properties: None,
+                required: None,
+                additional_properties: Some(false),
+                items: None,
+                any_of: None,
+            }
+        );
+
+        remove_multiple_test!(
+            false_multiple,
+            expected = Schema {
+                bson_type: Some(BsonType::Single(BsonTypeName::Object)),
+                properties: Some(map! {
+                    "x".into() => Schema::default(),
+                    "y".into() => Schema::default(),
+                }),
+                required: Some(vec!["x".into(), "y".into()]),
+                additional_properties: Some(false),
+                items: None,
+                any_of: None,
+            },
+            input = Schema {
+                bson_type: Some(BsonType::Multiple(vec![BsonTypeName::Object,])),
+                properties: Some(map! {
+                    "x".into() => Schema::default(),
+                    "y".into() => Schema::default(),
+                }),
+                required: Some(vec!["x".into(), "y".into()]),
+                additional_properties: Some(false),
+                items: None,
+                any_of: None,
+            }
+        );
+
+        remove_multiple_test!(
+            nullable_object,
```

Comment:
```suggestion
            object_and_other,
```
Arguably, the fact that the other type is "Null" does not matter. This test is just showing that only the "Object" gets the object-related fields, and the other type(s) do not.